pkg/datapath/iptables/iptables.go

```diff
@@ -694,6 +711,70 @@ func (m *IptablesManager) iptProxyRules(cmd string, proxyPort uint16, ingress bo
 	return nil
 }
 
+func noTrackRules(prog string, cmd string, IP string, port *lb.L4Addr, ingress bool) error {
+	protocol := strings.ToLower(port.Protocol)
+	p := strconv.FormatUint(uint64(port.Port), 10)
+	if ingress {
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "PREROUTING", "-p", protocol, "-d", IP, "--dport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "filter", cmd, "INPUT", "-p", protocol, "-d", IP, "--dport", p, "-j", "ACCEPT"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "OUTPUT", "-p", protocol, "-d", IP, "--dport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+	} else {
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "raw", cmd, "OUTPUT", "-p", protocol, "-s", IP, "--sport", p, "-j", "NOTRACK"}, false); err != nil {
+			return err
+		}
+		if _, err := runProgCombinedOutput(prog, []string{"-t", "filter", cmd, "OUTPUT", "-p", protocol, "-s", IP, "--sport", p, "-j", "ACCEPT"}, false); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func InstallNoTrackRules(IP string, port string, ipv6 bool) error {
+	prog := "iptables"
+	if ipv6 {
+		prog = "ip6tables"
+	}
+	ports, err := dnsPorts(port)
+	if err != nil {
+		return err
+	}
```

Comment:
This seems late for input validation, see [0]

Reply:
same as above